hibernate-reactive-panache-quickstart/src/main/java/org/acme/hibernate/orm/panache/FruitResource.java

```diff
@@ -0,0 +1,143 @@
+package org.acme.hibernate.orm.panache;
+
+import static javax.ws.rs.core.Response.Status.CREATED;
+import static javax.ws.rs.core.Response.Status.NOT_FOUND;
+import static javax.ws.rs.core.Response.Status.NO_CONTENT;
+
+import java.util.List;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+import javax.ws.rs.ext.Provider;
+
+import org.jboss.logging.Logger;
+import org.jboss.resteasy.reactive.RestPath;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import io.quarkus.hibernate.reactive.panache.Panache;
+import io.quarkus.panache.common.Sort;
+import io.smallrye.mutiny.CompositeException;
+import io.smallrye.mutiny.Uni;
+
+@Path("fruits")
+@ApplicationScoped
+@Produces("application/json")
+@Consumes("application/json")
+public class FruitResource {
+
+    private static final Logger LOGGER = Logger.getLogger(FruitResource.class.getName());
+
+    @GET
+    public Uni<List<Fruit>> get() {
+        return Fruit.listAll(Sort.by("name"));
+    }
+
+    @GET
+    @Path("{id}")
+    public Uni<Fruit> getSingle(@RestPath Long id) {
+        return Fruit.findById(id);
+    }
+
+    @POST
+    public Uni<Response> create(Fruit fruit) {
+        if (fruit == null || fruit.id != null) {
+            throw new WebApplicationException("Id was invalidly set on request.", 422);
+        }
+
+        return Panache.withTransaction(fruit::persist)
+                .onItem().transform(ignore -> Response.ok(fruit).status(CREATED).build());
+    }
+
+    @PUT
+    @Path("{id}")
+    public Uni<Response> update(@RestPath Long id, Fruit fruit) {
+        if (fruit == null || fruit.name == null) {
+            throw new WebApplicationException("Fruit name was not set on request.", 422);
+        }
+
+        return Panache
+                .withTransaction(() -> Fruit.<Fruit> findById(id)
+                        .onItem().ifNotNull().transform(entity -> {
+                            entity.name = fruit.name;
+                            return entity;
+                        }))
```

Comment:
Should use `invoke()` here.

Reply:
I need the entity so that I can add it to the response

Reply:
I suppose I could return the `Response` directly

Reply:
I will change it to this:
```
       return Panache
                .withTransaction(() -> Fruit.<Fruit> findById(id)
                        .onItem().ifNotNull().transform(entity -> {
                            entity.name = fruit.name;
                            return Response.ok(entity).build();
                        }))
                .onItem().ifNull().continueWith(() -> Response.ok().status(NOT_FOUND).build());
```

Reply:
But I think I prefer the other approach as it keeps the update of the entity and the return of the response separated